src/main/scala/org/constellation/infrastructure/endpoints/CheckpointEndpoints.scala

```diff
@@ -83,6 +98,62 @@ class CheckpointEndpoints[F[_]](implicit F: Concurrent[F], C: ContextShift[F]) e
       } yield response
   }
 
+  private[endpoints] def postFinishedCheckpoint(
+    checkpointBlockGossipService: CheckpointBlockGossipService[F],
+    messageValidator: MessageValidator,
+    snapshotService: SnapshotService[F],
+    checkpointService: CheckpointService[F],
+    checkpointAcceptanceService: CheckpointAcceptanceService[F]
+  ): HttpRoutes[F] =
+    HttpRoutes.of[F] {
+      case req @ POST -> Root / "peer" / "checkpoint" / "finished" =>
+        for {
+          message <- req.as[GossipMessage[CheckpointBlockPayload]]
+          payload = message.payload
+          senderId = req.headers.get(`X-Id`).map(_.value).map(Id(_)).get
+
+          res <- messageValidator.validateForForward(message, senderId) match {
+            case Invalid(EndOfCycle)                                                            => checkpointBlockGossipService.finishCycle(message) >> Ok()
+            case Invalid(IncorrectReceiverId(_, _)) | Invalid(PathDoesNotStartAndEndWithOrigin) => BadRequest()
+            case Invalid(IncorrectSenderId(_))                                                  => Response[F](status = Unauthorized).pure[F]
```

Comment:
Maybe we could log these three cases as well?

Reply:
I copied how it's done in gossip for snapshots. We can write a todo for this but I don't think it's necessary